### PR TITLE
Load Stocks Independently from Market Sync

### PR DESCRIPTION
**Pull request by patrick-hovsepian**
- load the portfolio file independently of syncing market data
- create ui directory/package and move renderer there
- rename project file to `main.py`
- remove the need for config.ini (will be more apparent why in next PR) -- all args get assigned a default and can be controlled directly through the CLI
- add initial buy/sell transaction support

### CODE REVIEW

File: main.py [hunk 1]
```diff
@@ -50,15 +44,19 @@ def main():
     verify_stock_keys(stocks_config)
     merge_config(config, args)
 
-    portfolio.populate(stocks_config, args)
+    # populate portfolio
+    portfolio = port.Portfolio()
+    portfolio.load_from_config(stocks_config)
+    portfolio.market_sync(args)
```
Comment (patrick-hovsepian) [inline main.py:50]: mainly this file was renamed but the heart of the change here is this

File: portfolio.py [hunk 2]
```diff
@@ -106,48 +181,8 @@ def average_buyin(self, buys: list, sells: list):
 
         return count, bought_at
 
-    # download all ticker data in a single request
-    # harder to parse but this provides a signficant performance boost
-    def download_market_data(self, args, stocks):
```
Comment (patrick-hovsepian) [inline portfolio.py:111]: the merge was a bit weird but this method is still the same, its just lower in the file

**Review by patrick-hovsepian — COMMENTED** [review]
highlights

**Comment by ConradSelig** [comment]
I like where this is headed - and with the context that an in-line command line will be present helps explain the removal of that config.ini. However, I'm thinking it might be smart to wait to pull this one until that pull request is ready - otherwise the only way to change the configuration is by command line argument, something that can be quite painful when changing a lot of the setting in config.ini.

Another question - will it be possible for users to set their own defaults? I personally don't want to have the change the width / height of the graph, the graphing mode, and the timezone everytime I run the program.

**Comment by patrick-hovsepian** [comment]
> to wait to pull this one until that pull request is ready

sure but its going to be a giant PR and wont be ready for a bit. i'll figure something out for defaults too by then

**Pull request closed without merging**